source/application/DbgHelp/DbgHelpTypes.h

```diff
@@ -1,33 +1,168 @@
-/*
+﻿/*
  * Copyright (c) 2016- Hourglass Resurrection Team
  * Hourglass Resurrection is licensed under GPL v2.
  * Refer to the file COPYING.txt in the project root.
  */
 
 #pragma once
 
-enum class DbgHelpArgType
+#include <string>
+#include <variant>
+
+using std::string_literals::operator""s;
+
+// This struct contains all basic types which have a defined size.
+// This doesn't include types like wchar_t without a defined size,
+// or types like long double for which no basic type exists in MSVC.
+class DbgHelpBasicType
 {
-    Unknown,
-    Char,
-    WideChar,
-    Char16,
-    Char32,
-    Short,
-    UnsignedShort,
-    Int,
-    UnsignedInt,
-    Long,
-    UnsignedLong,
-    LongLong,
-    UnsignedLongLong,
-    Float,
-    Double,
-    LongDouble,
-    /*
-     * These 3 are special, in the sense that they have "sub types".
-     */
-    Pointer,
-    Struct,
-    Union,
+public:
+    enum class BasicType {
+        Char,
+        Int8,
+        UnsignedInt8,
+        Int16,
+        UnsignedInt16,
+        Int32,
+        UnsignedInt32,
+        Int64,
+        UnsignedInt64,
+        Float,
+        Double,
+    };
+
+    const BasicType m_type;
+
+    DbgHelpBasicType(BasicType type) : m_type(type) {}
+
+    const std::wstring GetName() const
+    {
+        switch (m_type)
+        {
+        case BasicType::Char:
+            return L"char"s;
+        case BasicType::Int8:
+            return L"int8_t"s;
+        case BasicType::UnsignedInt8:
+            return L"uint8_t"s;
+        case BasicType::Int16:
+            return L"int16_t"s;
+        case BasicType::UnsignedInt16:
+            return L"uint16_t"s;
+        case BasicType::Int32:
+            return L"int32_t"s;

```

Comment:
How about replacing this stuff with `u8`, `i8`, `u16`, `i16` and so on so it takes up less space?